src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -184,21 +183,45 @@ public Mono<RequestAuthorizationResponseDto> requestTransactionAuthorization(
                                     .mapToInt(
                                             it.pagopa.ecommerce.commons.documents.v1.PaymentNotice::getAmount
                                     ).sum();
-                            return ecommercePaymentInstrumentsClient
-                                    .getPSPs(
-                                            amountTotal,
-                                            requestAuthorizationRequestDto.getLanguage().getValue(),
-                                            requestAuthorizationRequestDto.getPaymentInstrumentId()
+                            return ecommercePaymentInstrumentsClient // TODO gestione del carrello
+                                    .calculateFee(
+                                            new PaymentOptionDto()
+                                                    .paymentMethodId(
+                                                            requestAuthorizationRequestDto.getPaymentInstrumentId()
+                                                    )
+                                                    .touchpoint(transaction.getClientId().toString())
+                                                    .bin(
+                                                            requestAuthorizationRequestDto
+                                                                    .getDetails()instanceof CardAuthRequestDetailsDto cardData
+                                                                            ? cardData.getPan().substring(0, 8)
+                                                                            : null
+                                                    )
```

Comment:
Maybe an util method like `extractBin()` is better to use.

```
private Optional<string> extractBin(RequestAuthorizationRequestDetailsDto details) {
  //logic to extract bin
}

.bin(
                                                            extractBin().orElse("")
                                                    )
```

Reply:
orElse(""))must return a null value to create a GEC request what do you think ?